modules/images/image-loading-optimization/detection.php

```diff
@@ -39,16 +39,24 @@ function ilo_get_detection_script( string $slug, array $needed_minimum_viewport_
 	$web_vitals_lib_src  = add_query_arg( 'ver', $web_vitals_lib_data['version'], plugin_dir_url( __FILE__ ) . '/detection/web-vitals.js' );
 
 	$detect_args = array(
-		'serveTime'                   => microtime( true ) * 1000, // In milliseconds for comparison with `Date.now()` in JavaScript.
-		'detectionTimeWindow'         => $detection_time_window,
-		'isDebug'                     => WP_DEBUG,
-		'restApiEndpoint'             => rest_url( ILO_REST_API_NAMESPACE . ILO_URL_METRICS_ROUTE ),
-		'restApiNonce'                => wp_create_nonce( 'wp_rest' ),
-		'urlMetricsSlug'              => $slug,
-		'urlMetricsNonce'             => ilo_get_url_metrics_storage_nonce( $slug ),
-		'neededMinimumViewportWidths' => $needed_minimum_viewport_widths,
-		'storageLockTTL'              => ilo_get_url_metric_storage_lock_ttl(),
-		'webVitalsLibrarySrc'         => $web_vitals_lib_src,
+		'serveTime'               => microtime( true ) * 1000, // In milliseconds for comparison with `Date.now()` in JavaScript.
+		'detectionTimeWindow'     => $detection_time_window,
+		'isDebug'                 => WP_DEBUG,
+		'restApiEndpoint'         => rest_url( ILO_REST_API_NAMESPACE . ILO_URL_METRICS_ROUTE ),
+		'restApiNonce'            => wp_create_nonce( 'wp_rest' ),
+		'urlMetricsSlug'          => $slug,
+		'urlMetricsNonce'         => ilo_get_url_metrics_storage_nonce( $slug ),
+		'urlMetricsGroupStatuses' => array_map(
+			static function ( ILO_URL_Metrics_Group $group ): array {
+				return array(
+					'minimumViewportWidth' => $group->get_minimum_viewport_width(),
+					'complete'             => $group->is_complete(),
+				);
+			},
+			$group_collection->get_groups()
+		),
```

Comment:
+1, this puts this more specific logic only where needed without requiring the class based API to return that shape.